File: Easy/1752_Check_if_Array_Is_Sorted_and_Rotated/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        int count = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                count++;
            }
        }

        return count <= 1;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> nums1 = {3,4,5,1,2};
    cout << solution.check(nums1) << endl;

    // test cases 2
    vector<int> nums2 = {2,1,3,4};
    cout << solution.check(nums2) << endl;

    // test cases 3
    vector<int> nums3 = {1,2,3};
    cout << solution.check(nums3) << endl;

    return 0;
}

File: Easy/1752_Check_if_Array_Is_Sorted_and_Rotated/Solution.java


public class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        int count = 0;

        for (int i = 0; i < n; i++) {
            if (nums[i] > nums[(i + 1) % n]) {
                count++;
            }
        }

        return count <= 1;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {3, 4, 5, 1, 2};
        System.out.println(solution.check(nums1)); // true

        // test cases 2
        int[] nums2 = {2, 1, 3, 4};
        System.out.println(solution.check(nums2)); // false
        
        // test cases 3
        int[] nums3 = {1, 2, 3};
        System.out.println(solution.check(nums3)); // true
    }
}